### PR TITLE
Pass noReturn pragma to C code.

### PR DESCRIPTION
**Pull request by def-**
With GCC and Clang this generates `__attribute__((noreturn))` in the function declaration. (both tested) With VCC `__declspec(noreturn)` is used.

### CODE REVIEW

File: compiler/extccomp.nim [hunk 1]
```diff
@@ -27,7 +27,9 @@ type
     hasGcGuard,               # CC supports GC_GUARD to keep stack roots
     hasGnuAsm,                # CC's asm uses the absurd GNU assembler syntax
     hasNakedDeclspec,         # CC has __declspec(naked)
-    hasNakedAttribute         # CC has __attribute__((naked))
+    hasNakedAttribute,        # CC has __attribute__((naked))
```
Comment (Araq) [inline compiler/extccomp.nim]: I think it suffices to distinguish between __attribute__ and __declspec at this point.

Comment (def-) [inline compiler/extccomp.nim]: So I should just make it `hasAttribute` and `hasDeclspec` which includes compatibility for both naked and noreturn implicitly?

**Comment by def-** [comment]
This causes a problem with `nimkernel`. When you overwrite `panic` or `sysFatal` so that they return, then the `noreturn` pragma in `system/chcks.nim` is no longer true. Not sure how this should be fixed.

**Comment by def-** [comment]
I think it's alright now.

File: compiler/extccomp.nim [hunk 2]
```diff
@@ -26,8 +26,10 @@ type
     hasAssume,                # CC has __assume (Visual C extension)
     hasGcGuard,               # CC supports GC_GUARD to keep stack roots
     hasGnuAsm,                # CC's asm uses the absurd GNU assembler syntax
-    hasNakedDeclspec,         # CC has __declspec(naked)
-    hasNakedAttribute         # CC has __attribute__((naked))
+    hasDeclspec,              # CC has __declspec(X)
+    hasAttribute,             # CC has __attribute__((X))
+    hasNaked,                 # CC has naked declspec/attribute
+    hasNoreturn               # CC has noreturn declspec/attribute
```
Comment (Araq) [inline compiler/extccomp.nim]: That's not what I meant. Assume that ``hasDeclspec`` implies that both ``__declspec(naked)`` and ``__declspec(noreturn)`` are available. Likewise for ``hasAttribute``. I don't think there are compilers where this assumption does not hold.

Comment (def-) [inline compiler/extccomp.nim]: Alright, here we go.

**Comment by yglukhov** [comment]
This PR has introduced a regression: #2696.

**Pull request merged**